WalletWasabi/Blockchain/Analysis/BlockchainAnalyzer.cs

```diff
@@ -105,7 +105,23 @@ private void AnalyzeCoinjoin(SmartTransaction tx, int newInputAnonset, ISet<HdPu
 			.WalletOutputs.GroupBy(x => x.Amount)
 			.ToDictionary(x => x.Key, y => y.Count());
 
-		var anonsets = tx.Transaction.GetAnonymitySets(tx.WalletOutputs.Select(x => x.Index));
+		var indistinguishableOutputs = tx.Transaction.Outputs
```

Comment:
You could move this whole new code to NBitcoinExtensions.GetAnonymitySets but with an override of

public static IDictionary<uint, int> GetAnonymitySets(this Transaction me, IEnumerable<uint> outputIndices, WalletInputs, WalletOutputs);

The idea is to have similar logics in the same place and to lighten AnalyzeCoinjoin method.